plugin/src/main/java/org/opensearch/ml/model/MLModelManager.java

```diff
@@ -609,7 +609,7 @@ private void uploadModel(MLRegisterModelInput registerModelInput, MLTask mlTask,
 
     private void registerModelFromUrl(MLRegisterModelInput registerModelInput, MLTask mlTask, String modelVersion) {
         String taskId = mlTask.getTaskId();
-        FunctionName functionName = mlTask.getFunctionName();
+        FunctionName functionName = registerModelInput.getFunctionName();
```

Comment:
Any reason to change to `registerModelInput` ? Will it break BWC?

Reply:
If we don't provide url, the function name from ML task would be read from request body, while the modelInput would be generated from config. If we still use the function name from ml task, the name would still be null and the default would be "text_embedding". I have tried locally for body with url and without url, both worked for me.

Reply:
MLTask must track the correct function name. Can you check if the function name in MLTask correct or not?

Reply:
No. I have tested it again. If the request body is like:
{
  "name": "amazon/neural-sparse/opensearch-neural-sparse-encoding-v1",
  "version": "1.0.0",
  "model_format": "TORCH_SCRIPT"
}

The function name inside the ml task would be text_embedding.

Reply:
I mean, after changing to `registerModelInput.getFunctionName();`, is the function name in MLTask correct now?

Reply:
Already done by set function name inside mltask and rewrite to ml index.